proposals/propertyDependencies.md

```diff
@@ -0,0 +1,292 @@
+those goals means that some trade-offs need to be made. {{alternatives}} lists
+some alternatives that we considered.
+
+## A Vocabulary for Applying Subschemas
+
+This document adds the `propertyDependencies` keyword to the
+`https://json-schema.org/vocab/applicator` [applicator
+vocabulary](../jsonschema-core.html#applicatorvocab).
+
+### `propertyDependencies`
+
+This keyword specifies subschemas that are evaluated if the instance is an
+object and contains a certain property with a certain string value.
+
+This keyword's value MUST be an object. Each value in the object MUST be an
+object whose values MUST be valid JSON Schemas.
+
+If the outer object key is a property in the instance and the inner object key
+is equal to the value of that property, the entire instance must validate
+against the schema. Its use is dependent on the presence and value of the
+property.
+
+Omitting this keyword has the same behavior as an empty object.
+
+## [Appendix] Problems With Existing Patterns {#problems}
+
+### `oneOf`/`anyOf`
+
+The pattern of using `oneOf` to describe a choice between two schemas has become
+ubiquitous.
+
+```jsonschema
+{
+  "oneOf": [
+    { "$ref": "#/$defs/aaa" },
+    { "$ref": "#/$defs/bbb" }
+  ]
+}
+```
+
+However, this pattern has several shortcomings. The main problem is that it
+tends to produce confusing error messages. Some implementations employ
+heuristics to guess the user's intent and provide better messaging, but that's
+not wide-spread or consistent behavior, nor is it expected or required from
+implementations.
+
+This pattern is also inefficient. Generally, there is a single value in the
+object that determines which alternative to chose, but the `oneOf` pattern has
+no way to specify what that value is and therefore needs to evaluate the entire
+schema. This is made worse in that every alternative needs to be fully validated
+to ensure that only one of the alternative passes and all the others fail. This
+last problem can be avoided by using `anyOf` instead, but that pattern is much
+less used.
+
+### `if`/`then`
+
+We can describe this kind of constraint more efficiently and with with better
+error messaging by using `if`/`then`. This allows the user to explicitly specify
+the constraint to be used to select which alternative the schema should be used
+to validate the schema. However, this pattern has problems of it's own. It's
+verbose, error prone, and not particularly intuitive, which leads most people to
+avoid it.
+
+```jsonschema
+{
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "foo": { "const": "aaa" }
+        },
+        "required": ["foo"]
+      },
+      "then": { "$ref": "#/$defs/foo-aaa" }
+    },
+    {
+      "if": {
+        "properties": {
+          "foo": { "const": "bbb" }
+        },
+        "required": ["foo"]
+      },
+      "then": { "$ref": "#/$defs/foo-bbb" }
+    }
+  ]
+}
+```
+
+## [Appendix] Alternatives Considered {#alternatives}
```

Comment:
I think the alternatives need to be listed in the same area as the limitations (which arguably need to stay with the solution).  This should be moved back up.